### ISSUE TITLE
loss waves too much

### ISSUE THREAD

**Issue by yuimo** [OpenNMT/OpenNMT-tf#38]
hi, i'm using transformer model to train my chinese to english model, 
the config is below
after training for 140K batches, the loss still changes very much, and the value of loss looks like too big for other tasks. 
any ideas? thanks.
<img width="954" alt="image" src="https://user-images.githubusercontent.com/22741826/33749667-f620691a-dc09-11e7-8cdc-1781067a732d.png">

  3 params:
  4   optimizer: GradientDescentOptimizer
  5   learning_rate: 0.1
  6   param_init: 0.1
  7   clip_gradients: 5.0
  8   decay_type: exponential_decay
  9   decay_rate: 0.7
 10   decay_steps: 5000
 11   start_decay_steps: 150000
 12   beam_width: 4
 13   maximum_iterations: 250
 14   external_evaluators: BLEU
 15 train:
 16   batch_size: 100
 17   save_checkpoints_steps: 5000
 18   save_summary_steps: 50
 19   train_steps: 1000000
 20   eval_delay: 7200 # Every 5 hours.
 21   maximum_features_length: 50
 22   maximum_labels_length: 50
 23   keep_checkpoint_max: 200
 24   save_eval_predictions: True
 25 infer:
 26   batch_size: 30

**Comment by guillaumekln** [comment]
Hello,

I suppose you are using the latest commit. If so, the cross entropy is no more normalized in time by default. This is the behavior of other OpenNMT versions for example. You should look at the `normalized_loss` curve instead for a token-level loss.

You can still disable this behavior and normalize the cross entropy in every dimensions by setting the parameter:

```yaml
params:
  average_loss_in_time: true
```

**Comment by guillaumekln** [comment]
I'm closing this one. If you have any more questions, feel free to ping me.

**Comment by huache** [comment]
Actually, I encounter same problem about one month ago, and I assumed that the transformer model of `OpenNMT-tf` is not maturely enough to use. 

I recommend adding this explanation to document, and I'd like to help improving the document of `OpenNMT-tf` if I could provide any help.

**Comment by guillaumekln** [comment]
Did the loss wave even when you configured it to be averaged in time?

**Comment by huache** [comment]
I just using the trained model to translate some text, and compare it with translation of tensor2tensor. The BLEU score of OpenNMT-tf is little higher. 
I will try to use that parameter tomorrow, then report it.

**Comment by huache** [comment]
Hi, I added "average_loss_in_time: true" in config file, and trained a transformer model again. This is the loss graph:
![transformer-avg-loss](https://user-images.githubusercontent.com/11642845/34756374-5b788d50-f605-11e7-94d4-3de1fce20f9e.png)

It looks much better, but not smooth enough. May be it caused by the decay strategy of learning rate ?

In the paper "Attention is all your need", they use a special strategy to change the learning rate, but I don't know how to reproduce it by using `OpenNMT-tf`, could you give me some clue ?

**Comment by guillaumekln** [comment]
See the configuration file [config/optim/adam_with_noam_decay.yml](https://github.com/OpenNMT/OpenNMT-tf/blob/master/config/optim/adam_with_noam_decay.yml).

**Comment by huache** [comment]
Wow !  I will try it as soon as possible, thanks a lot !
